flexmeasures/data/models/planning/__init__.py

```diff
@@ -34,26 +36,38 @@ class Scheduler:
     __version__ = None
     __author__ = None
 
-    sensor: Sensor
+    sensor: Optional[Sensor] = None
+    asset: Optional[Asset] = None
+
+    asset_or_sensor: Asset | Sensor
+
     start: datetime
     end: datetime
     resolution: timedelta
     belief_time: datetime
+
     round_to_decimals: int
+
     flex_model: Optional[dict] = None
     flex_context: Optional[dict] = None
+
     fallback_scheduler_class: "Type[Scheduler] | None" = None
     info: dict | None = None
 
     config_deserialized = False  # This flag allows you to let the scheduler skip checking config, like timing, flex_model and flex_context
     return_multiple = False
 
+    # set to True if the Scheduler supports triggering on an Asset or False
+    # if the Scheduler expects a Sensor
+    requires_asset = False
```

Comment:
This confused me. Consider renaming to `supports_scheduling_an_asset`. The `Scheduler` class does support both, it seems. But I get that it should be `False`, for backwards compatibility with schedulers that subclassed our `Scheduler` and do not yet support scheduling an asset. That could be made clearer.